lib/core/src/lib/common/services/user-preferences.service.ts

```diff
@@ -246,11 +247,15 @@ export class UserPreferencesService {
 
     private getLanguageByKey(key: string): LanguageItem {
         const defaultLanguage = { key: 'en' } as LanguageItem;
+        let language: LanguageItem;
 
-        const registeredLanguages = this.appConfig.get<Array<LanguageItem>>(AppConfigValues.APP_CONFIG_LANGUAGES_KEY);
-        if (registeredLanguages && Array.isArray(registeredLanguages)) {
-            return registeredLanguages.find((language) => key.includes(language.key)) || defaultLanguage;
+        const customLanguages = this.appConfig.get<Array<LanguageItem>>(AppConfigValues.APP_CONFIG_LANGUAGES_KEY);
+        if (customLanguages && Array.isArray(customLanguages)) {
+            language = customLanguages.find((language) => key.includes(language.key));
         }
-        return defaultLanguage;
+        if (!language) {
+            language = DEFAULT_LANGUAGE_LIST.find((language) => language.key === key) ?? defaultLanguage;
```

Comment:
instead of if we can write it a bit shorter: 
```ts
language ??= DEFAULT_LANGUAGE_LIST.find((language) => language.key === key) ?? defaultLanguage;
```
It will assign right part only when language is null or undefined, see doc: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment

Reply:
That looks much cleaner now, thanks!!